packages/devtools_app/lib/src/profiler/flutter/cpu_profile_flame_chart.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:flutter/material.dart';
+
+import '../../charts/flutter/flame_chart.dart';
+import '../../ui/colors.dart';
+import '../../utils.dart';
+import '../cpu_profile_model.dart';
+
+class CpuProfileFlameChart extends FlameChart<CpuProfileData, CpuStackFrame> {
+  CpuProfileFlameChart(
+    CpuProfileData data, {
+    @required double width,
+    @required CpuStackFrame selected,
+    @required Function(CpuStackFrame stackFrame) onSelected,
+  }) : super(
+          data,
+          duration: data.profileMetaData.time.duration,
+          totalStartingWidth: width,
+          startInset: sideInsetSmall,
+          endInset: sideInsetSmall,
+          selected: selected,
+          onSelected: onSelected,
+        );
+
+  @override
+  _CpuProfileFlameChartState createState() => _CpuProfileFlameChartState();
+}
+
+class _CpuProfileFlameChartState
+    extends FlameChartState<CpuProfileFlameChart, CpuStackFrame> {
+  static const stackFramePadding = 1;
+
+  int _colorOffset = 0;
+
+  @override
+  void initFlameChartElements() {
+    super.initFlameChartElements();
+    expandRows(widget.data.cpuProfileRoot.depth +
+        rowOffsetForTopPadding +
+        rowOffsetForBottomPadding);
+
+    final Map<String, double> stackFrameLefts = {};
+
+    double leftForStackFrame(CpuStackFrame stackFrame) {
```

Comment:
maybe make these private members in the class instead of inner methods?

Reply:
done